NMS-14614: Quick Start - Notifications (#5204)
OpenNMS/opennms: docs/modules/operation/pages/quick-start/notifications.adoc

= Step 2: Turn on Default Notifications

> This section describes how to turn on default notifications in {page-component-title}.

Author: @mmahacek, could you clarify how notifications are treated by default once they're turned on?

> This section describes how to turn on default notifications in {page-component-title}.

Reviewer: There's a handful of notification definitions shipped out of the box for node/interface/service up/down events as well as for high/low thresholds. Not all are enabled by default (still need to enable notices on threshold UEIs after enabling the notification system)

> This section describes how to turn on default notifications in {page-component-title}.

Reviewer 2: Adding to what @mmahacek has already mentioned, interfaceDown, nodeDown, nodeLostService, nodeAdded, interfaceDeleted are turned on out-of-the-box. There are others but they need to be turned on manually. They are all set to destination path Email-Admin which the target is Admin group. The default method is javaEmail. As long as the javamail properties are set correctly, email will be sent.

{page-component-title} uses notifications to inform users about events in the network without requiring them to log in to the web UI.

> Notifications are commonly sent through email and pager messages.

Reviewer: ```suggestion
Notifications are commonly sent through email and text/pager messages, but can be customized to send to a variety of external systems.
```

> For additional information, including alternative notification mechanisms, see xref:operation:deep-dive/notifications/introduction.adoc[Notifications] in the Deep Dive section.

Reviewer: ```suggestion
For additional information, including alternative notification mechanisms, see xref:operation:deep-dive/notifications/introduction.adoc[Notifications] in the Deep Dive section and the xref:reference:daemons/daemon-config-files/notifd.adoc[notifd daemon] reference section.
```

== Enable notifications

Notifications are disabled by default in a fresh {page-component-title} installation.
Follow these steps to enable notifications:

. Log in as a user with admin privileges.
. Click the *Gear* icon in the top-right of the screen.
. Under Event Management, click *On* beside Notification Status, and click *Update*.

image::notifications/notifications-on.png[notifications-on, 300]

The green bell icon indicates that notifications have been enabled globally.

> NOTE: The workflow above is functionally equivalent to editing `notifd-configuration.xml` and setting `status="on"` in the top-level `notifd-configuration` element.

Reviewer: ```suggestion
NOTE: The workflow above is functionally equivalent to editing `$\{OPENNMS_HOME}/etc/notifd-configuration.xml` and setting `status="on"` in the top-level `notifd-configuration` element.
```

This configuration file update happens immediately, with no need to restart {page-component-title} or send an event.

> == Configure destination paths

Reviewer: A Destination Path is required for a notification.  There's a default one created that emails the "Admin" user.

> == Configure destination paths

Reviewer 3: So we could just have them use the default to the Admin user, if we tell them to update the Admin user email in Step 1. Since we recommend that they NOT use the default Admin user, perhaps I should update the Step 1: First-time Login chapter to have them create their own user, then we can have them specify a destination path in the Notifications section that uses the user they specified in Step 1.

A destination path specifies the "who" (recipient), "when" (interval), and "how" (method) of a notification.
It also specifies escalation targets.
Configuring destination paths separately from individual events lets you use the same information for multiple notifications.
This minimizes duplication and encourages reuse.

Follow these steps to configure a destination path:

. Log in as a user with admin privileges.
. Click the *Gear* icon in the top-right of the screen, and choose menu:Event Management[Configure Notifications > Configure Destination Paths].
. Click *New Path*, fill in the desired information, and click *Finish*.
** To add a destination path, select it from the list in the *Destinations Path* screen, and click *Edit*.

> By default, {page-component-title} runs the destination path of all notifications matching the event's UEI.
> You can configure {page-component-title} to run only the destination path of the first matching notification by editing `notifd-configuration.xml` and setting `match-all="false"` in the top-level `notifd-configuration` element.

Reviewer: ```suggestion
It is possible to define multiple notifications for the same UEI to use different destination paths.
By default, {page-component-title} runs the destination path of all notifications matching the event's UEI.
You can configure notifd to process only the destination path of the first matching notification by editing `$\{OPENNMS_HOME}/etc/notifd-configuration.xml` and setting `match-all="false"` in the top-level `notifd-configuration` element.
```

> NOTE: Configuration of destination paths is stored in `destinationPaths.xml`.

Reviewer: ```suggestion
NOTE: Configuration of destination paths is stored in `$\{OPENNMS_HOME}/etc/destinationPaths.xml`.
```

Changes to this file happen immediately, with no need to restart {page-component-title} or send an event.

== Configure event notifications

> An event notification ties any {page-component-title} event to a destination path.

Reviewer: ```suggestion
An event notification defines a destination path and message template for a specific event UEI.
It is possible to define multiple notifications for the same UEI to utilize different filters, destination paths, and/or message templates.
```

Configure event notifications by following these steps:

. Log in as a user with admin privileges.
. Click the *Gears* icon in the top-right of the screen, and choose menu:Event Management[Configure Notifications > Configure Event Notifications].
The screen displays a list of events that are configured for notifications.
. Click *Add New Event Notification*.
. (Optional) Build a rule to match a subset of IP addresses or services, and click menu:Validate Rule Results[Next] or *Skip Results Validation*.
. In the *Choose Path* screen, specify information to be included in the notification, including the destination path, subject (for an email message), body text, and so on.
. Click *Finish*.

> NOTE: Event notification configuration is stored in `notifications.xml`.

Reviewer: ```suggestion
NOTE: Event notification configuration is stored in `$\{OPENNMS_HOME}/etc/notifications.xml`.
```

Changes to this file happen immediately, with no need to restart {page-component-title} or send an event.

=== Filter rules

> The filter rule configured in `notifications.xml` (for example, `<rule>IPADDR != '0.0.0.0'</rule>`) is not strict by default.

Reviewer: ```suggestion
The filter rule configured in `$\{OPENNMS_HOME}/etc/notifications.xml` (for example, `<rule>IPADDR != '0.0.0.0'</rule>`) is not strict by default.
```

= Step 2: Turn on Default Notifications

This section describes how to turn on default notifications in {page-component-title}.

{page-component-title} uses notifications to inform users about events in the network without requiring them to log in to the web UI.
Notifications are commonly sent through email and text/pager messages, but can be customized to send to a variety of external systems.
For additional information, including alternative notification mechanisms, see xref:operation:deep-dive/notifications/introduction.adoc[Notifications] in the Deep Dive section and the xref:reference:daemons/daemon-config-files/notifd.adoc[notifd daemon] reference section.

== Enable notifications

Notifications are disabled by default in a fresh {page-component-title} installation.
Follow these steps to enable notifications:

. Log in as a user with admin privileges.

> . Click the *Gear* icon in the top-right of the screen.

Author: ```suggestion
. Click the *Gear* symbol in the top-right of the screen.
```

. Under Event Management, click *On* beside Notification Status, and click *Update*.

image::notifications/notifications-on.png[notifications-on, 300]

> The green bell icon at the top-left of the screen indicates that notifications are enabled globally.

Author: ```suggestion
The green bell symbol at the top-left of the screen indicates that notifications are enabled globally.
```

* interfaceDeleted (on by default)
* interfaceDown (on by default)
* nodeAdded (on by default)
* nodeDown (on by default)
* nodeLostService (on by default)

You can turn them on and off in the menu:Event Management[Configure Notifications > Configure Event Notifications] screen.
See xref:operation:deep-dive/notifications/introduction.adoc[Notifications] in the Deep Dive section for more information.

== Configure destination paths

A destination path specifies the "who" (recipient), "when" (delay), and "how" (method) of a notification.
It also specifies escalation targets.
Configuring destination paths separately from individual events lets you use the same information for multiple notifications.
This minimizes duplication and encourages reuse.

You must specify a destination path to receive notifications from {page-component-title}.
The `Email-Admin` destination path is defined by default, and is the notification path when no other destination paths are specified.
It sends email notifications to users in the Admin group (see xref:operation:quick-start/users.adoc#ga-users-next[Users]).

Follow these steps to define a new destination path:

. Log in as a user with admin privileges.

> . Click the *Gear* icon in the top-right of the screen, and choose menu:Event Management[Configure Notifications > Configure Destination Paths].

Author: ```suggestion
. Click the *Gear* symbol in the top-right of the screen, and choose menu:Event Management[Configure Notifications > Configure Destination Paths].
```

This minimizes duplication and encourages reuse.

You must specify a destination path to receive notifications from {page-component-title}.
The `Email-Admin` destination path is defined by default, and is the notification path when no other destination paths are specified.
It sends email notifications to users in the Admin group (see xref:operation:quick-start/users.adoc#ga-users-next[Users]).

Follow these steps to define a new destination path:

. Log in as a user with admin privileges.
. Click the *Gear* icon in the top-right of the screen, and choose menu:Event Management[Configure Notifications > Configure Destination Paths].
. Click *New Path* and configure it:
.. Specify a name for the path (for example, Email-Bob) and define an initial delay.
.. Click *Edit*, and select the users, groups, roles, or email addresses to send notifications to.
... To add a single email address to the Send to Email Addresses list, click *Add Address*, type the email address, and click *OK*.
... To remove email addresses from the Send to Email Addresses list, select them from the list and click *Remove Selected Addresses*.
.. Click *Next Step*.

> .. (Optional) Define a delay interval.

Reviewer 3: ```suggestion
.. (Optional) Define a delay interval for group notification.
```
Added this information, because I don't get this option when I click "Next Step". It appears only for a group.

This will determine the delay between sending notifications to each member of a group.
+
For example, defining a five minute delay will mean that, when a notification is generated, the first member of the group will be notified immediately.

> Subsequent members will then be notified in five minute increments, from the top of the list of group members to the bottom.

Reviewer 3: ```suggestion
Subsequent members will then be notified in five-minute increments, from the top of the list of group members to the bottom.
```
